internal/rbd/rbd_attach.go

```diff
@@ -462,9 +464,24 @@ func detachRBDImageOrDeviceSpec(
 	}
 	if dArgs.isNbd && dArgs.logDir != "" {
 		logFile := getCephClientLogFileName(dArgs.volumeID, dArgs.logDir, "rbd-nbd")
-		if err = os.Remove(logFile); err != nil {
-			log.WarningLog(ctx, "failed to remove logfile: %s, error: %v",
-				logFile, err)
+		switch logStrategyLookup[dArgs.logStatergy] {
```

Comment:
this switch adds quite some complexity to the existing function, move it to a helper that handles the log strategy? Also makes it easier to write a unit-test for that.